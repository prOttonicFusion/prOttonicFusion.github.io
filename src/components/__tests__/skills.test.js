import React from 'react'
import { render } from '@testing-library/react'
import { jest } from '@jest/globals'
import Skills from '../skills'

jest.mock('../../sections/skills.mdx', () => {
    return {
        __esModule: true,
        default: () => {
            return <div>Some content</div>
        },
    }
})

describe('Skills section', () => {
    it('renders skills correctly', () => {
        const { container } = render(
            <Skills />,
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
