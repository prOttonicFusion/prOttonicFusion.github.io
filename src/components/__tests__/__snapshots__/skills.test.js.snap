// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills section renders skills correctly 1`] = `
<div
  id="skills"
>
  <div>
    <div>
      Some content
    </div>
  </div>
  <div
    class="skills-container"
  >
    <div
      class="skill-category"
    >
      <div
        class="skill-category-title"
      >
        Languages
      </div>
      <div
        class="category-skills-container"
      >
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              Python
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 100%;"
            />
          </div>
        </div>
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              JavaScript/NodeJS
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 100%;"
            />
          </div>
        </div>
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              C++
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 65%;"
            />
          </div>
        </div>
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              Fortran95
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 80%;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="skill-category"
    >
      <div
        class="skill-category-title"
      >
        Full-Stack
      </div>
      <div
        class="category-skills-container"
      >
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              React Native
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 85%;"
            />
          </div>
        </div>
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              HTML + CSS
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 100%;"
            />
          </div>
        </div>
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              Qt
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 60%;"
            />
          </div>
        </div>
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              Databases/SQL
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 85%;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="skill-category"
    >
      <div
        class="skill-category-title"
      >
        Platforms
      </div>
      <div
        class="category-skills-container"
      >
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              Docker
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 75%;"
            />
          </div>
        </div>
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              Unix/Linux
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 100%;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="skill-category"
    >
      <div
        class="skill-category-title"
      >
        High-Performance Computing
      </div>
      <div
        class="category-skills-container"
      >
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              OpenMPI
            </span>
            <span
              class="skill-badge-container"
            >
              <img
                alt="Dev icon"
                class="dev-icon"
                height="15px"
                src="test-file-stub"
              />
            </span>
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 75%;"
            />
          </div>
        </div>
        <div
          class="skill-item"
        >
          <div
            class="skill-description-container"
          >
            <span
              class="skill-title"
            >
              LAMMPS
            </span>
            <span
              class="skill-badge-container"
            />
          </div>
          <div
            class="progress-bar"
          >
            <span
              class="progress"
              style="width: 95%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
